cumulus/parachains/runtimes/coretime/coretime-westend/src/genesis_config_presets.rs

```diff
@@ -0,0 +1,103 @@
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Coretime Westend Runtime genesis config presets
+
+use crate::*;
+use alloc::{vec, vec::Vec};
+use cumulus_primitives_core::ParaId;
+use frame_support::build_struct_json_patch;
+use parachains_common::{AccountId, AuraId};
+use sp_genesis_builder::PresetId;
+use sp_keyring::Sr25519Keyring;
+use testnet_parachains_constants::westend::{
+	currency::UNITS as WND, xcm_version::SAFE_XCM_VERSION,
+};
+
+const CORETIME_WESTEND_ED: Balance = ExistentialDeposit::get();
+
+fn coretime_westend_genesis(
+	invulnerables: Vec<(AccountId, AuraId)>,
+	endowed_accounts: Vec<AccountId>,
+	endowment: Balance,
+	id: ParaId,
+) -> serde_json::Value {
+	build_struct_json_patch!(RuntimeGenesisConfig {
+		balances: BalancesConfig {
+			balances: endowed_accounts.iter().cloned().map(|k| (k, endowment)).collect(),
+		},
+		parachain_info: ParachainInfoConfig { parachain_id: id },
+		collator_selection: CollatorSelectionConfig {
+			invulnerables: invulnerables.iter().cloned().map(|(acc, _)| acc).collect(),
+			candidacy_bond: CORETIME_WESTEND_ED * 16,
+		},
+		session: SessionConfig {
+			keys: invulnerables
+				.into_iter()
+				.map(|(acc, aura)| {
+					(
+						acc.clone(),          // account id
+						acc,                  // validator id
+						SessionKeys { aura }, // session keys
+					)
+				})
+				.collect(),
+		},
+		polkadot_xcm: PolkadotXcmConfig { safe_xcm_version: Some(SAFE_XCM_VERSION) },
+	})
+}
+
+/// Provides the JSON representation of predefined genesis config for given `id`.
+pub fn get_preset(id: &PresetId) -> Option<Vec<u8>> {
+	let patch = match id.as_ref() {
+		sp_genesis_builder::LOCAL_TESTNET_RUNTIME_PRESET => coretime_westend_genesis(
+			// initial collators.
+			vec![
+				(Sr25519Keyring::Alice.to_account_id(), Sr25519Keyring::Alice.public().into()),
+				(Sr25519Keyring::Bob.to_account_id(), Sr25519Keyring::Bob.public().into()),
+			],
+			Sr25519Keyring::well_known().map(|k| k.to_account_id()).collect(),
+			WND * 1_000_000,
+			1000.into(),
+		),
+		sp_genesis_builder::DEV_RUNTIME_PRESET => coretime_westend_genesis(
+			// initial collators.
+			vec![(Sr25519Keyring::Alice.to_account_id(), Sr25519Keyring::Alice.public().into())],
+			vec![
+				Sr25519Keyring::Alice.to_account_id(),
+				Sr25519Keyring::Bob.to_account_id(),
+				Sr25519Keyring::AliceStash.to_account_id(),
+				Sr25519Keyring::BobStash.to_account_id(),
+			],
+			WND * 1_000_000,
+			1000.into(),
```

Comment:
wouldn't it be better to keep the `CORETIME_PARA_ID` here? i.e. 1005?

Reply:
Yes, I have to apologize, this is a copy-paste error. 🙈 Thanks for the detailed review.

Reply:
Actually, I know now why the bug got introduced, I was careless in copy-pasting because I thought we only define the para-id on the top-level in the polkadot-parachain: https://github.com/paritytech/polkadot-sdk/issues/7618.

But this is a future issue, for know I will just align the values again.

Reply:
Fixed in bcdc0fdf31ef6f791d8bc52618943a86c5ade80b